data_structures/trie_using_hashmap.cpp

```diff
@@ -0,0 +1,339 @@
+/**
+ * @file
+ * @author [Venkata Bharath](https://github.com/bharath000)
+ * @brief Implementation of [Trie](https://en.wikipedia.org/wiki/Trie) data
+ * structure using HashMap for different characters and method for predicting
+ * words based on prefix.
+ * @details The Trie data structure is implemented using unordered map to use
+ * memory optimally, predict_words method is developed to recommend words based
+ * on a given prefix along with other methods insert, delete, search, startwith
+ * in trie.
+ * @see trie_modern.cpp for difference
+ */
+#include <cassert>        /// for assert
+#include <iostream>       /// for IO operations
+#include <memory>         /// for std::shared_ptr
+#include <stack>          /// for std::stack
+#include <unordered_map>  /// for std::unordered_map
+#include <vector>         /// for std::vector
+
+/**
+ * @namespace data_structures
+ * @brief Data structures algorithms
+ */
+namespace data_structures {
+/**
+ *@brief Trie class. implementaion of trie using hashmap in each trie node for
+ *all the charters of char16_t(UTF-16)type with methods to insert, delete,
+ *search, startwith and to recommend words based on a given prefix.
+ */
+class Trie {
+ private:
+    /**
+     *  @brief struct representing nodes in trie
```

Comment:
```suggestion
     * @brief struct representing nodes in the trie
```

Reply:
Thank you for helping me, I have updated the documentation as requested. I can see that the workflow has been updated recently, Will this have impact on my commit?

Reply:
> Will this have an impact on my commit

No, everything should work perfectly fine. 🙂